File: 241001/행복한 수열의 개수/number-of-happy-sequence.cpp
#include <iostream>
#include <algorithm>
using namespace std;
#define MAX_NM 100

int n, m;
int grid[MAX_NM][MAX_NM];

int main() {
    // 여기에 코드를 작성해주세요.
    cin >> n>>m;
    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            cin >> grid[i][j];
        }
    }

    int ans = 0;

    for(int i=0; i<n; i++) {
        int cnt = 1;
        int max_cnt = 0;
        for(int j=1; j<n; j++) {
            if(grid[i][j-1] == grid[i][j]) {
                cnt++;
                max_cnt = max(max_cnt, cnt);
            } else {
                cnt = 1;
            }
        }

        if (max_cnt >= m) ans++;
    }

    for(int i=0; i<n; i++) {
        int cnt = 1;
        int max_cnt = 0;
        for(int j=1; j<n; j++) {
            if(grid[j-1][i] == grid[j][i]) {
                cnt++;
                max_cnt = max(max_cnt, cnt);
            } else {
                cnt = 1;
            }
        }

        if (max_cnt >= m) ans++;
    }

    cout << ans;
    return 0;
}

File: 241031/행복한 수열의 개수/number-of-happy-sequence.py
n, m = tuple(map(int, input().split()))
grid = [
    list(map(int, input().split()))
    for _ in range(n)
]

ans = 0

for i in range(n):
    cnt = 1
    max_cnt = 1

    for j in range(n-1):
        if grid[i][j] == grid[i][j+1]:
            cnt += 1
        else:
            max_cnt = max(max_cnt, cnt)
            cnt = 1
    
    max_cnt = max(max_cnt, cnt)
    if max_cnt >= m:
        ans += 1

    #print(ans)
    cnt = 1
    max_cnt = 1

    for j in range(n-1):
        if grid[j][i] == grid[j+1][i]:
            cnt += 1
        else:
            max_cnt = max(max_cnt, cnt)
            cnt = 1
    
    max_cnt = max(max_cnt, cnt)
    if max_cnt >= m:
        ans += 1

print(ans)